internal/trace/context.go

```diff
@@ -26,11 +26,22 @@ type contextKey int
 // loggerKey is the associated key type for logger entry in context.
 const loggerKey contextKey = iota
 
-// WithLoggerLevel returns a context with logrus log entry.
-func WithLoggerLevel(ctx context.Context, level logrus.Level) (context.Context, logrus.FieldLogger) {
+// NewLogger returns a logger.
+func NewLogger(debug bool, verbose bool) (context.Context, logrus.FieldLogger) {
+
+	var logLevel logrus.Level
+	if debug {
+		logLevel = logrus.DebugLevel
+	} else if verbose {
+		logLevel = logrus.InfoLevel
+	} else {
+		logLevel = logrus.WarnLevel
+	}
```

Comment:
This isn't parsing command line options, it is taking two bools and making them a logrus value. Command line parsing should not know about logrus and this creates that separation of concerns.

Reply:
> Command line parsing should not know about logrus

This makes sense. 


Since the `context` must come from CLI, the function should be `func WithLogger(ctx context. Context, debug bool, verbose bool) context. Context` and caller gets logger by calling the following `Logger` on-demand.

Reply:
Yeh it doesn't fit what people would expect from a New or a With right now because of the context. If I handle that, it should fit the New pattern. The With pattern is normally used for what are essentially optional parameters to object creation.

Reply:
> The With pattern is normally used for what are essentially optional parameters to object creation.

Makes sense. Let's merge this after 1.0.0 release cut-off and I will check if the `NewXXX` pattern fits into context management in oras 1.1.0.